g3doc/tutorials/graph_keras_cnn_flowers.ipynb

```diff
@@ -0,0 +1,1198 @@
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "RLRxxV4G7McQ"
+      },
+      "source": [
+        "graph_reg_history = graph_reg_model.fit(\n",
+        "    train_dataset,\n",
+        "    validation_data=validation_dataset,\n",
+        "    epochs=HPARAMS.train_epochs,\n",
+        "    callbacks=[tf.keras.callbacks.EarlyStopping(patience=5)])"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "QyLIa0qEi74A"
+      },
+      "source": [
+        "### Plot training progress of the graph-regularized model"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "84ffIFmd7PMz"
+      },
+      "source": [
+        "graph_reg_history_dict = graph_reg_history.history\n",
+        "\n",
+        "acc = graph_reg_history_dict['accuracy']\n",
+        "val_acc = graph_reg_history_dict['val_accuracy']\n",
+        "loss = graph_reg_history_dict['loss']\n",
+        "graph_loss = graph_reg_history_dict['scaled_graph_loss']\n",
+        "val_loss = graph_reg_history_dict['val_loss']\n",
+        "\n",
+        "epochs = range(1, len(acc) + 1)\n",
+        "\n",
+        "plt.clf()   # clear figure\n",
+        "\n",
+        "# \"-r^\" is for solid red line with triangle markers.\n",
+        "plt.plot(epochs, loss, '-r^', label='Training loss')\n",
+        "# \"-gD\" is for solid green line with diamond markers.\n",
+        "plt.plot(epochs, graph_loss, '-gD', label='Training graph loss')\n",
+        "# \"-b0\" is for solid blue line with circle markers.\n",
+        "plt.plot(epochs, val_loss, '-bo', label='Validation loss')\n",
+        "plt.title('Training and validation loss')\n",
+        "plt.xlabel('Epochs')\n",
+        "plt.ylabel('Loss')\n",
+        "plt.legend(loc='best')\n",
+        "\n",
+        "plt.show()"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "5EJUJJja7lEB"
+      },
+      "source": [
+        "plt.clf()   # clear figure\n",
+        "\n",
+        "plt.plot(epochs, acc, '-r^', label='Training acc')\n",
+        "plt.plot(epochs, val_acc, '-bo', label='Validation acc')\n",
+        "plt.title('Training and validation accuracy')\n",
+        "plt.xlabel('Epochs')\n",
+        "plt.ylabel('Accuracy')\n",
+        "plt.legend(loc='best')\n",
+        "\n",
+        "plt.show()"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "cShQNXGNjIOs"
+      },
+      "source": [
+        "Based on the performances of the two models (with and without graph regularization) we observe that the model performs better without graph regularization. But an ablation study (on the reduced projection dimensionality) reveals a different trend\n",
```

Comment:
This is not a good message for prospective users. Can we we set the projection dimensionality to a value that gives maximum improvement for graph regularization?